Fails to build on powerpc

```
     Running `CARGO=/usr/local/bin/cargo CARGO_CRATE_NAME=dua CARGO_MANIFEST_DIR=/wrkdirs/usr/ports/sysutils/dua-cli/work/dua-cli-2.14.2 CARGO_PKG_AUTHORS='Sebastian Thiel <byronimo@gmail.com>' CARGO_PKG_DESCRIPTION='A tool to conveniently learn about the disk usage of directories, fast'\!'' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=dua-cli CARGO_PKG_REPOSITORY='https://github.com/Byron/dua-cli' CARGO_PKG_VERSION=2.14.2 CARGO_PKG_VERSION_MAJOR=2 CARGO_PKG_VERSION_MINOR=14 CARGO_PKG_VERSION_PATCH=2 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps:/usr/local/lib' /usr/local/bin/rustc --crate-name dua --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C panic=abort -C linker-plugin-lto -C codegen-units=1 --cfg 'feature="crosstermion"' --cfg 'feature="open"' --cfg 'feature="tui"' --cfg 'feature="tui-crossplatform"' --cfg 'feature="tui-react"' --cfg 'feature="tui-shared"' --cfg 'feature="unicode-segmentation"' -C metadata=8e77ad818c1efb78 -C extra-filename=-8e77ad818c1efb78 --out-dir /wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps -L dependency=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps --extern anyhow=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libanyhow-bbb6f89415c51efc.rmeta --extern atty=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libatty-e802f37d66aba940.rmeta --extern byte_unit=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libbyte_unit-cab8261ef7bfae8d.rmeta --extern clap=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libclap-74ab6aba082fc1c2.rmeta --extern colored=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libcolored-ccb6deb0911988ca.rmeta --extern crosstermion=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libcrosstermion-94e83ec96c8555d3.rmeta --extern filesize=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libfilesize-bb98e33f22bc2fb0.rmeta --extern itertools=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libitertools-92e1cbdb87b12e77.rmeta --extern jwalk=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libjwalk-930c238f234a6c0d.rmeta --extern num_cpus=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libnum_cpus-7bfdc87e6db4858d.rmeta --extern open=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libopen-e362b6c678fcf8a0.rmeta --extern petgraph=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libpetgraph-43c9d81153cb63a7.rmeta --extern tui=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libtui-f1d6b5aa5b209ce3.rmeta --extern tui_react=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libtui_react-c04972b17b8af4e5.rmeta --extern unicode_segmentation=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libunicode_segmentation-989a8d636ae0851e.rmeta --extern wild=/wrkdirs/usr/ports/sysutils/dua-cli/work/target/release/deps/libwild-f8dc786536775329.rmeta -C linker=cc -C link-arg=-fstack-protector-strong`
error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
  --> src/aggregate.rs:11:18
   |
11 |         atomic::{AtomicU64, Ordering},
   |                  ^^^^^^^^^
   |                  |
   |                  no `AtomicU64` in `sync::atomic`
   |                  help: a similar name exists in the module: `AtomicU8`

error[E0599]: no method named `load` found for struct `Arc<_>` in the current scope
  --> src/aggregate.rs:50:38
   |
50 |                         shared_count.load(Ordering::Acquire)
   |                                      ^^^^ method not found in `Arc<_>`
   |
   = note: `shared_count` is a function, perhaps you wish to call it

error[E0599]: no method named `fetch_add` found for struct `Arc<_>` in the current scope
  --> src/aggregate.rs:65:26
   |
65 |             shared_count.fetch_add(1, Ordering::Relaxed);
   |                          ^^^^^^^^^ method not found in `Arc<_>`
   |
   = note: `shared_count` is a function, perhaps you wish to call it

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `dua-cli`
```

This is because there are no 64-bit atomics for powerpc in Rust.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,13 +10,12 @@ license = "MIT"
 include = ["src/**/*", "Cargo.*", "LICENSE", "README.md", "CHANGELOG.md", "!**/tests/*"]
 
 [features]
-default = ["tui-crossplatform", "trash-move", "aggregate-scan-progress"]
+default = ["tui-crossplatform", "trash-move"]
 tui-unix = ["crosstermion/tui-react-termion", "tui-shared"]
 tui-crossplatform = ["crosstermion/tui-react-crossterm", "tui-shared"]
 
 tui-shared = ["tui", "tui-react", "open", "unicode-segmentation"]
 trash-move = ["trash"]
-aggregate-scan-progress = []
 
 [dependencies]
 clap = { version = "3.0", features = ["derive"] }

--- a/Makefile
+++ b/Makefile
@@ -38,7 +38,6 @@ check:## run cargo-check with various features
 	cargo check --no-default-features --features tui-unix
 	cargo check --no-default-features --features tui-crossplatform
 	cargo check --no-default-features --features trash-move
-	cargo check --no-default-features --features aggregate-scan-progress
 
 unit-tests: ## run all unit tests
 	cargo test --all

--- a/src/aggregate.rs
+++ b/src/aggregate.rs
@@ -3,24 +3,71 @@ use anyhow::Result;
 use colored::{Color, Colorize};
 use filesize::PathExt;
 use std::{borrow::Cow, io, path::Path};
-#[cfg(feature = "aggregate-scan-progress")]
 use std::{
     sync::{
-        atomic::{AtomicUsize, Ordering},
+        atomic::{AtomicBool, Ordering},
         Arc,
     },
     thread,
     time::Duration,
 };
 
+/// Throttle access to an optional `io::Write` to the specified `Duration`
+#[derive(Debug)]
+struct ThrottleWriter<W> {
+    out: Option<W>,
+    trigger: Arc<AtomicBool>,
+}
+
+impl<W> ThrottleWriter<W>
+where
+    W: io::Write,
+{
+    fn new(out: Option<W>, duration: Duration) -> Self {
+        let writer = Self {
+            out,
+            trigger: Default::default(),
+        };
+
+        if writer.out.is_some() {
+            let trigger = Arc::downgrade(&writer.trigger);
+            thread::spawn(move || {
+                thread::sleep(Duration::from_secs(1));
+                while let Some(t) = trigger.upgrade() {
+                    t.store(true, Ordering::Relaxed);
+                    thread::sleep(duration);
+                }
+            });
+        }
+
+        writer
+    }
+
+    fn throttled<F>(&mut self, f: F)
+    where
+        F: FnOnce(&mut W),
+    {
+        if self.trigger.swap(false, Ordering::Relaxed) {
+            self.unthrottled(f);
+        }
+    }
+
+    fn unthrottled<F>(&mut self, f: F)
+    where
+        F: FnOnce(&mut W),
+    {
+        if let Some(ref mut out) = self.out {
+            f(out);
+        }
+    }
+}
+
 /// Aggregate the given `paths` and write information about them to `out` in a human-readable format.
 /// If `compute_total` is set, it will write an additional line with the total size across all given `paths`.
 /// If `sort_by_size_in_bytes` is set, we will sort all sizes (ascending) before outputting them.
 pub fn aggregate(
     mut out: impl io::Write,
-    #[cfg_attr(not(feature = "aggregate-scan-progress"), allow(unused_variables))] err: Option<
-        impl io::Write + Send + 'static,
-    >,
+    err: Option<impl io::Write>,
     walk_options: WalkOptions,
     compute_total: bool,
     sort_by_size_in_bytes: bool,
@@ -35,28 +82,7 @@ pub fn aggregate(
     let mut num_roots = 0;
     let mut aggregates = Vec::new();
     let mut inodes = InodeFilter::default();
-
-    #[cfg(feature = "aggregate-scan-progress")]
-    let shared_count = Arc::new(AtomicUsize::new(0));
-
-    #[cfg(feature = "aggregate-scan-progress")]
-    if let Some(mut out) = err {
-        thread::spawn({
-            let shared_count = Arc::clone(&shared_count);
-            move || {
-                thread::sleep(Duration::from_secs(1));
-                loop {
-                    thread::sleep(Duration::from_millis(100));
-                    write!(
-                        out,
-                        "Enumerating {} entries\r",
-                        shared_count.load(Ordering::Acquire)
-                    )
-                    .ok();
-                }
-            }
-        });
-    }
+    let mut progress = ThrottleWriter::new(err, Duration::from_millis(100));
 
     for path in paths.into_iter() {
         num_roots += 1;
@@ -65,8 +91,9 @@ pub fn aggregate(
         let device_id = crossdev::init(path.as_ref())?;
         for entry in walk_options.iter_from_path(path.as_ref()) {
             stats.entries_traversed += 1;
-            #[cfg(feature = "aggregate-scan-progress")]
-            shared_count.fetch_add(1, Ordering::Relaxed);
+            progress.throttled(|out| {
+                write!(out, "Enumerating {} entries\r", stats.entries_traversed).ok();
+            });
             match entry {
                 Ok(entry) => {
                     let file_size = match entry.client_state {
@@ -99,6 +126,9 @@ pub fn aggregate(
                 Err(_) => num_errors += 1,
             }
         }
+        progress.unthrottled(|out| {
+            write!(out, "\x1b[2K\r").ok();
+        });
 
         if sort_by_size_in_bytes {
             aggregates.push((path.as_ref().to_owned(), num_bytes, num_errors));

